cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriter.java

```diff
@@ -77,16 +79,10 @@ public DatasetSystemMetadataWriter(MetadataStore metadataStore,
     try (SystemDatasetInstantiator dsInstantiator = dsInstantiatorFactory.create();
          Dataset dataset = dsInstantiator.getDataset(dsInstance)) {
       if (dataset instanceof RecordScannable) {
```

Comment:
Since being `RecordWritable` does not guarantee a Dataset to be explorable, we thought we'll leave it out for now.